### ISSUE TITLE
matchAll: false positive of regexp/no-unused-capturing-group

### ISSUE THREAD

**Issue by panmenghan** [ota-meshi/eslint-plugin-regexp#399]
```javascript
const a = [...'var s1="s1"\nvar s2="s2"'.matchAll(/var (s\d+)=("[^"]+")/g)].map(
  m => [m[1], m[2]]
)
```
